### PR TITLE
MULE-16872: Async in a transaction executes in the same thread as the

### PR DESCRIPTION
**Pull request by elrodro83**
transaction

### CODE REVIEW

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java [hunk 1]
```diff
@@ -119,6 +121,41 @@ public void withinSubflowDoesntUseFlowMaxConcurrency() throws Exception {
...
+    final Startable withSourceTx =
+        (Startable) (locator.find(Location.builderFromStringRepresentation("with-source-tx").build())).get();
+    withSourceTx.start();
+
+    assertThat(queueHandler.read("asyncDispatched", 1000), not(nullValue()));
+    assertThat(queueHandler.read("asyncRunning", 1000), not(nullValue()));
+  }
+
+  @Test
+  @Description("Assert that async blocks run outside of the transaction from the caller flow")
```
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/routing/AsyncTestCase.java]: from the try?

File: integration/src/test/resources/org/mule/test/routing/async-test.xml [hunk 2]
```diff
@@ -77,5 +79,76 @@
...
+    <flow name="with-source-tx" initialState="stopped">
+        <tx:transactional-source config-ref="localTx" transactionalAction="ALWAYS_BEGIN"/>
+
+        <script:execute engine="groovy" target="latch">
+            <script:code>
+                new java.util.concurrent.CountDownLatch(1)
+            </script:code>
+        </script:execute>
+    
+        <logger message="Before async... (#[payload])" />
```
Comment (afelisatti) [inline integration/src/test/resources/org/mule/test/routing/async-test.xml]: Remove loggers.

File: integration/src/test/java/org/mule/test/routing/AsyncTestCase.java [hunk 3]
```diff
@@ -119,6 +121,41 @@ public void withinSubflowDoesntUseFlowMaxConcurrency() throws Exception {
...
+  public void withTryTx() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    runFlows("with-try-tx", latch);
```
Comment (afelisatti) [inline integration/src/test/java/org/mule/test/routing/AsyncTestCase.java]: Why don't these need starting?

Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/AsyncTestCase.java]: the flowRunner already starts the flow if needed

Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/AsyncTestCase.java]: https://github.com/mulesoft/mule/blob/mule-4.x/tests/component-plugin/src/main/java/org/mule/functional/api/flow/FlowRunner.java#L76

**Pull request merged**